src/main/java/org/wildfly/security/pem/Pem.java

```diff
@@ -190,4 +229,17 @@ public static void generatePemX509Certificate(ByteStringBuilder target, X509Cert
             throw log.certificateParseError(e);
         }
     }
+
+    /**
+     * Generate PEM content containing a {@link PublicKey}.
+     *
+     * @param target the target byte string builder (must not be {@code null})
+     * @param publicKey the {@link PublicKey} (must not be {@code null})
+     */
+    public static void generatePemPublicKey(ByteStringBuilder target, PublicKey publicKey) {
+        Assert.checkNotNullParam("target", target);
+        Assert.checkNotNullParam("publicKey", publicKey);
+
+        generatePemContent(target, "PUBLIC KEY", ByteIterator.ofBytes(publicKey.getEncoded()));
```

Comment:
For now, I think we should handle X.509 encoding format only. That would be enough to address the requirements for the token-based realm. In this case, we can introduce a check for that.

Should we create a JIRA to support other encoding formats ?